File: examples/function/deploy/node/deployAFunction.js
const { FunctionService } = require("m3o/function");

// Deploy a group of functions
async function deployAfunction() {
  let functionService = new FunctionService(process.env.MICRO_API_TOKEN);
  let rsp = await functionService.deploy({
    entrypoint: "helloworld",
    name: "my-first-func",
    project: "tests",
    repo: "github.com/m3o/nodejs-function-example",
    runtime: "nodejs14",
  });
  console.log(rsp);
}

deployAfunction();

File: examples/function/deploy/go/deployAFunction.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/function"
)

// Deploy a group of functions
func DeployAfunction() {
	functionService := function.NewFunctionService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := functionService.Deploy(&function.DeployRequest{
		Entrypoint: "helloworld",
		Name:       "my-first-func",
		Project:    "tests",
		Repo:       "github.com/m3o/nodejs-function-example",
		Runtime:    "nodejs14",
	})
	fmt.Println(rsp, err)
}
